src/pages/docs/tutorials/tutorial-typescript-enact/component-with-ts-enact/index.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Add a New Component Using TypeScript and Enact
+github: https://github.com/enactjs/docs/blob/develop/src/pages/docs/tutorials/tutorial-typescript-basic/component-with-ts-enact/index.md
+order: 4
+---
+
+### Counter Component using TypeScript with Enact
+
+We can use the same **Counter.tsx** file, remove the previous content and add the following contents:
+
+```ts
+//Counter.tsx
+
+import * as React from 'react';
+import Button from '@enact/moonstone/Button';
+import kind from '@enact/core/kind';
+
+const CounterBase = kind({
+    name: 'Counter',
+
+    defaultProps: {
+        count: 0
+    },
+
+    render: ({onIncrementClick, onDecrementClick, onResetClick, count, ...rest}) => (
+        <div>
+            <h1>{count}</h1>
+            <Button onClick={onDecrementClick}>Decrement --</Button>
+            <Button onClick={onResetClick}>Reset</Button>
+            <Button onClick={onIncrementClick}>Increment ++</Button>
+        </div>
+    )
+});
+
+export default CounterBase;
+```
+
+### View the Counter in the Browser
+
+![Enact Typescript Counter](Typescript_Enact_view.png)
+
+The above code holds the definition of the `Counter` component. Using `kind()` we are binding `defaultProps`, `render` and all the events together.
+
+> Please check API documentation in the Core Library to know more about [kind](../../../modules/core/kind/) and [handle](../../../modules/core/handle/)
+
+We will add `handler` inside the `kind` declaration to handle the click event on the buttons:
+
+```ts
+    handlers: {
+        onDecrementClick: createHandler(count => count - 1),
+        onIncrementClick: createHandler(count => count + 1),
+        onResetClick: createHandler(() => 0)
+    },
+```
+
+- Use `ui/Changeable` for state management of `count`
+
+```ts
+import Changeable from '@enact/ui/Changeable';
+```
+
+ > Applying `Changeable` to a component will pass two additional props: the current value from state and an event callback to invoke when the value changes. For more information, read the [ui/Changeable documentation](../../../modules/ui/Changeable/)
+
+```ts
+const Counter = Changeable({prop: 'count' , change: 'onCounterChange'}, CounterBase);
+
+```
+
+- Change the default export to the new `Counter` component:
+
+```ts
+export default Counter;
+```
+
+- Create a handle function for click events on the button. The `createHandler` function will take a function as input then use the function to update the `count`. By using `handle` we will forward the call to the callback function (`onCounterChange`) defined in `Changeable`:
+
+```ts
+function createHandler(fn) {
+    return handle(
```

Comment:
We’re missing the import for `adaptEvent`, `forward`, and `handle` at this point of the example.